src/topology/fanout.rs

```diff
@@ -129,7 +129,30 @@ impl Sink for Fanout {
                 Ok(Async::NotReady) => {
                     all_complete = false;
                 }
-                Err(()) => self.handle_sink_error()?,
+                Err(()) => self.handle_sink_error(i)?,
+            }
+        }
+
+        if all_complete {
+            Ok(Async::Ready(()))
+        } else {
+            Ok(Async::NotReady)
+        }
+    }
+
+    fn close(&mut self) -> Poll<(), Self::SinkError> {
```

Comment:
It seems the only difference between `close` and `poll_complete` is `match sink.close()` vs `match sink.poll_complete()`. Is there any way we can DRY this up a bit?